src/app/hooks/useTxErrorMessage.ts

```diff
@@ -0,0 +1,14 @@
+import { useTranslation } from 'react-i18next'
+import { TxError } from '../../oasis-nexus/api'
+
+export const useTxErrorMessage = (error: TxError | undefined): string | undefined => {
+  const { t } = useTranslation()
+  if (!error) return undefined
+  if (error.module === 'evm' && error.code === 8 && !error.message) {
```

Comment:
One alternative would be to just do this with translations. For example:

t(`errors.${error.module}.${error.code}${error.message ?? ''}`)

And the key would be `errors.evm.8` -> poiniting to custom error.

I imagine that there will be other edge cases, that would need this and it would look nicer with `switch` and direct translation implementation.

Reply:
The possibility of other corner cases has been discussed at length on Slack, and the conclusion was there is probably nothing else, and if there is, we want to handle it in Nexus.